### PR TITLE
chore: enable prefer-object-spread

### PR DESCRIPTION
**Pull request by abrahamguo**
## PR Checklist

- [x] Addresses an existing open issue: fixes part of #9523 
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Enable `prefer-object-spread`. All reports were auto-fixable. ([docs](https://eslint.org/docs/latest/rules/prefer-object-spread))

### CODE REVIEW

File: packages/eslint-plugin/src/rules/indent.ts [hunk 1]
```diff
@@ -317,7 +320,7 @@ export default createRule<Options, MessageIds>({
         });
       },
 
-      TSIndexedAccessType(node: TSESTree.TSIndexedAccessType) {
+      TSIndexedAccessType(node: TSESTree.TSIndexedAccessType): void {
```
Comment (abrahamguo) [inline packages/eslint-plugin/src/rules/indent.ts:323]: `@typescript-eslint/explicit-function-return-type` began reporting on these once we stopped using `Object.assign`.

**Review by kirkwaiblinger — DISMISSED** [review]

**Comment by JoshuaKGoldberg** [comment]
👋 @abrahamguo you've sent us a lot of great PRs recently and we want to send you some money as thanks (https://typescript-eslint.io/maintenance/contributor-tiers#community-reimbursements). I can't for the life of me find a personal email for you. Could you please email me at the email on my profile? 😄

**Review by JoshuaKGoldberg — DISMISSED** [review]
_heck yes_

**Review by Josh-Cena — APPROVED** [review]
Awesome work

**Review by Josh-Cena — APPROVED** [review]

**Pull request merged**